core/src/com/unciv/ui/cityscreen/CitizenManagementTable.kt

```diff
@@ -34,7 +35,13 @@ class CitizenManagementTable(val cityScreen: CityScreen) : Table() {
         val avoidCell = Table()
         avoidCell.touchable = Touchable.enabled
         avoidCell.add(avoidLabel).pad(5f)
-        avoidCell.onClick { city.avoidGrowth = !city.avoidGrowth; city.reassignPopulation(); cityScreen.update() }
+        avoidCell.onClick {
+            if (!UncivGame.Current.gameInfo.currentPlayerCiv.isSpectator() && !city.isPuppet) {
```

Comment:
Instead of checking isSpectator(), use cityScreen.canChangeState
There's also several other buttons you'll need to disable